lib/counter.js

```diff
@@ -23,22 +23,24 @@ var getLabels = require('./util').getLabels;
  * @param {Array.<string>} labels - Labels
  * @constructor
  */
-
-
 function Counter(name, help, labels) {
 	var config;
 	if(isObject(name)) {
 		config = extend({
 			labelNames: [],
-			registers: [ register ]
 		}, name);
+		if(name.registers !== undefined) {
+			config.registers = name.registers;
+		} else {
+			config.registers = [ globalRegistry ];
```

Comment:
When looking into util-extend's code, it wasn't completely clear if it will try to merge arrays or replace them. With closer look, it seems to be replacing them in which case we can use it. I will do the changes.